views/js/runner/proxy/cache/proxy.js

```diff
@@ -63,27 +64,37 @@ define([
             var self = this;
 
             //we keep items here
-            this.itemStore    = itemStoreFactory(cacheSize);
-
-            //keep track of the calls to prevent doing it again on navigating back and forward
-            this.nextCalledBy = {};
+            this.itemStore = itemStoreFactory(cacheSize);
 
             //can we load the next item from the cache/store ?
-            this.getItemFromStore  = false;
-
-            //check whether we're on the last item
-            this.isLastItem   = false;
+            this.getItemFromStore = false;
 
             //keep a ref to the promise of the loadNextItem in case the call is not done when moving
             this.loadNextPromise = Promise.resolve();
 
+            //preload at least this number of items
+            this.cacheAmount = 1;
+
+            //keep reference on the test map as we don't have access to the test runner
+            this.testMap = null;
+            this.on('receive', function (data) {
+                if (data) {
+                    if (data.testData && data.testData.config && data.testData.config.itemCaching) {
+                        self.cacheAmount = parseInt(data.testData.config.itemCaching.amount, 10) || self.cacheAmount;
+                    }
+                    if (data.testMap) {
+                        self.testMap = data.testMap;
+                    }
+                }
+            });
+
             /**
              * Update the item state in the store
              * @param {String} uri - the item identifier
-             * @param {Objetc} state - the state of the item
+             * @param {Object} state - the state of the item
              * @returns {Boolean}
              */
-            this.updateState = function updateState(uri, state){
+            this.updateState = function updateState(uri, state) {
```

Comment:
Could you please rename `uri` by `id` or `itemIdentifier` everywhere to avoid confusion.